lib/open_food_network/products_cache_refreshment.rb

```diff
@@ -17,31 +17,27 @@
 module OpenFoodNetwork
   class ProductsCacheRefreshment
     def self.refresh(distributor, order_cycle)
-      unless pending_job? distributor, order_cycle
-        enqueue_job distributor, order_cycle
-      end
+      job = refresh_job(distributor, order_cycle)
+      enqueue_job(job) unless pending_job?(job)
     end
 
+    class << self
+      private
 
-    private
-
-    def self.pending_job?(distributor, order_cycle)
-      # To inspect each job, we need to deserialize the payload.
-      # This is slow, and if it's a problem in practice, we could pre-filter in SQL
-      # for handlers matching the class name, distributor id and order cycle id.
+      def refresh_job(distributor, order_cycle)
+        RefreshProductsCacheJob.new(distributor.id, order_cycle.id)
+      end
 
-      Delayed::Job.
-        where(locked_at: nil).
-        map(&:payload_object).
-        select { |j|
-          j.class == RefreshProductsCacheJob &&
-          j.distributor_id == distributor.id &&
-          j.order_cycle_id == order_cycle.id
-        }.any?
```

Comment:
Yes, the logic is the same. We check for an identical job.

*BUT:* The previous implementation needed to call `payload_object` on each job. That call fails if the object can't be deserialised properly. That happens when the payload object is a user and that user doesn't exist any more. In that case `map(&:payload_object)` raised `Delayed::DeserializationError` caused by `ActiveRecord::RecordNotFound`. Instead of rescueing from that error in a separate each block, I simplified the implementation to just check for the serialised content in the db column `handler`. That avoids the error and is more efficient.